addons/pos_paytm/models/pos_payment_method.py

```diff
@@ -0,0 +1,97 @@
+# coding: utf-8
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import logging
+import requests
+
+from odoo.addons.pos_paytm import utils as paytm_utils
+from odoo.exceptions import UserError
+from odoo import fields, models, api, _
+from datetime import datetime
+from dateutil import tz
+
+_logger = logging.getLogger(__name__)
+
+class PosPaymentMethod(models.Model):
+    _inherit = 'pos.payment.method'
+
+    paytm_mid = fields.Char(string='PayTM Merchant ID')
+    paytm_tid = fields.Char(string='PayTM Terminal ID')
+    merchant_key = fields.Char(string='PayTM Merchant Key')
+    channel_id = fields.Char(string='PayTM Channel ID', default='EDC')
+    paytm_test_mode = fields.Boolean(string='Test Mode', help='Run transactions in the test environment.')
+    accept_payment = fields.Selection(selection=[('auto', 'Automatically'), ('manual', 'Manually')], default='auto')
+    allowed_payment_modes = fields.Selection(selection=[('all', 'All'), ('card', 'Card'), ('qr', 'QR')], default='all')
+
+    def _get_payment_terminal_selection(self):
+        selection = super()._get_payment_terminal_selection()
```

Comment:
set a variable for this is useless
`return super()._get_payment_terminal_selection() + [('paytm', 'PayTM')]`